src/app/cras/edit-cra/edit-cra.component.ts

```diff
@@ -0,0 +1,88 @@
+import { Component, OnInit } from '@angular/core';
+import { Router, ActivatedRoute, Params, UrlSegment } from '@angular/router';
+
+import { SerializerService } from 'src/app/shared/serialization/serializer.service';
+import { Cra } from 'src/app/shared/cra.model';
+
+@Component({
+  selector: 'app-edit-cra',
+  templateUrl: './edit-cra.component.html',
+  styleUrls: ['./edit-cra.component.scss']
+})
+export class EditCraComponent implements OnInit {
+
+  cra = new Cra();
+  editToken: string;
+  reviewToken: string;
+
+  saved = false;
+  showModal = false;
+
+  title = {
+    add: 'Créer',
+    edit: 'Editer',
+    review: 'Consulter',
+  };
+
+  mode: string;
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private serializer: SerializerService
+  ) {}
+
+  ngOnInit() {
+
+    this.route.queryParams.subscribe(
+      (parms: Params) => {
+        if (parms.hasOwnProperty('data')) {
+          const datas = <{mode: string, cra: Cra}>this.serializer.deserialize(parms['data']);
+          this.mode = datas.mode;
+          this.cra = <Cra>datas.cra;
```

Comment:
With my previous comment, you don't need to put `<Cra` and TS know the `datas.cra` type